tests/unit/components/menu/GitBranchMenu.spec.js

```diff
@@ -0,0 +1,153 @@
+  FetchEvent: {
+    subscribe: jest.fn(),
+  },
+}));
+
+jest.mock('src/composables/Project', () => ({
+  getCurrentBranch: jest.fn(() => Promise.resolve('main')),
+  getBranches: jest.fn(() => Promise.resolve([])),
+  fetchGit: jest.fn(() => Promise.resolve()),
+}));
+
+describe('Test component: GitBranchMenu', () => {
+  let wrapper;
+  let subscribe;
+  let unsubscribe;
+
+  useRoute.mockImplementation(() => ({
+    params: {
+      projectName: 'project-00000000',
+      viewType: 'model',
+    },
+  }));
+
+  beforeEach(() => {
+    subscribe = jest.fn();
+    unsubscribe = jest.fn();
+    GitEvent.FetchEvent.subscribe.mockImplementation(() => {
+      subscribe();
+      return { unsubscribe };
+    });
+    wrapper = shallowMount(GitBranchMenu, {
+      props: {
+        currentBranchName: 'main',
+      },
+      mocks: {
+        DialogEvent,
+      },
+      global: {
+        plugins: [
+          createI18n(i18nConfiguration),
+        ],
+      },
+    });
+  });
+
+  describe('Test functions', () => {
+    describe('Test function: openCloseExpandMenu', () => {
+      it('Should invert local value', () => {
+        expect(wrapper.vm.showLocal).toEqual(false);
+        expect(wrapper.vm.showRemote).toEqual(false);
+
+        wrapper.vm.openCloseExpandMenu(true);
+        expect(wrapper.vm.showLocal).toEqual(true);
+        expect(wrapper.vm.showRemote).toEqual(false);
+
+        wrapper.vm.openCloseExpandMenu(true);
+        expect(wrapper.vm.showLocal).toEqual(false);
+        expect(wrapper.vm.showRemote).toEqual(false);
+      });
+
+      it('Should invert remote value', () => {
+        expect(wrapper.vm.showLocal).toEqual(false);
+        expect(wrapper.vm.showRemote).toEqual(false);
+
+        wrapper.vm.openCloseExpandMenu(false);
+        expect(wrapper.vm.showLocal).toEqual(false);
+        expect(wrapper.vm.showRemote).toEqual(true);
+
+        wrapper.vm.openCloseExpandMenu(false);
+        expect(wrapper.vm.showLocal).toEqual(false);
+        expect(wrapper.vm.showRemote).toEqual(false);
+      });
+    });
+
+    describe('Test function: initBranches', () => {
+      it('Should be empty on no branches', async () => {
+        getBranches.mockImplementation(() => Promise.resolve([]));
+        await wrapper.vm.initBranches();
+        expect(wrapper.vm.branches).toEqual({ local: [], remote: [] });
+      });
+
+      it('Local branch array should contains all local branches', async () => {
+        getBranches.mockImplementation(() => Promise.resolve([
+          new Branch({
+            name: 'Local',
+            onLocal: true,
+          }),
+        ]));
+        await wrapper.vm.initBranches();
+        expect(wrapper.vm.branches).toEqual({
+          local: [
+            new Branch({
+              name: 'Local',
+              onLocal: true,
+            }),
+          ],
+          remote: [],
+        });
+      });
+
+      it('Remote branch array should contains all local branches', async () => {
```

Comment:
`Remote branch array should contains all local branches` => `should fill remote branch array with all remote branches`

Reply:
done